connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -0,0 +1,338 @@
+}
+
+async function formatMessagesForUpsert(channelId: string, messages: Message[]) {
+  return (
+    await Promise.all(
+      messages.map(async (message) => {
+        const text = await processMessageForMentions(message.text as string);
+
+        const userName = await getUserName(message.user as string);
+        const messageDate = new Date(parseInt(message.ts as string, 10) * 1000);
+        const messageDateStr = formatDateForUpsert(messageDate);
+
+        return `>> @${userName} [${messageDateStr}]:\n ${text}\n`;
+      })
+    )
+  ).join("\n");
+}
+
+async function upsertToDatasource(
+  dataSourceConfig: DataSourceConfig,
+  documentId: string,
+  documentContent: string
+) {
+  console.log(`Upserting to Dust data source ${documentId}`, documentContent);
+  const dust_url = `${FRONT_API}/api/v1/w/${dataSourceConfig.workspaceId}/data_sources/${dataSourceConfig.dataSourceName}/documents/${documentId}`;
+  const dust_request_payload = {
+    text: documentContent,
+  };
+  const dust_request_config: AxiosRequestConfig = {
+    headers: {
+      Authorization: `Bearer ${dataSourceConfig.workspaceAPIKey}`,
+    },
+  };
+  const dust_request_result = await axios.post(
+    dust_url,
+    dust_request_payload,
+    dust_request_config
+  );
+
+  if (dust_request_result.status >= 200 && dust_request_result.status < 300) {
+    return;
+  } else {
+    throw new Error(
+      `Failed to upsert to Dust data source ${dataSourceConfig.dataSourceName}: ${dust_request_result.status} ${dust_request_result.statusText}`
+    );
+  }
+}
+
+export async function fetchUsers(slackAccessToken: string) {
+  let cursor: string | undefined;
+  const client = new WebClient(slackAccessToken);
+  do {
+    const res = await client.users.list({});
+    if (res.error) {
+      throw new Error(`Failed to fetch users: ${res.error}`);
+    }
+    if (!res.members) {
+      throw new Error(`Failed to fetch users: members is undefined`);
+    }
+    for (const member of res.members) {
+      if (member.id && member.real_name) {
+        cacheSet(getUserCacheKey(member.id), member.real_name);
+      }
+    }
+    cursor = res.response_metadata?.next_cursor;
+  } while (cursor);
+}
+
+export async function getAccessToken(
+  nangoConnectionId: string
+): Promise<string> {
+  if (!NANGO_SLACK_CONNECTOR_ID) {
+    throw new Error("NANGO_SLACK_CONNECTOR_ID is not defined");
+  }
+  return nango_client().getToken(NANGO_SLACK_CONNECTOR_ID, nangoConnectionId);
+}
+
+export async function saveSuccessSyncActivity(
+  dataSourceConfig: DataSourceConfig
+) {
+  const connector = await Connector.findOne({
+    where: {
+      type: "slack",
+      workspaceId: dataSourceConfig.workspaceId,
+      dataSourceName: dataSourceConfig.dataSourceName,
+    },
+  });
+  if (!connector) {
+    throw new Error(
+      `Could not find the connectors to mark it as success :/ ${dataSourceConfig.workspaceId} ${dataSourceConfig.dataSourceName}`
+    );
+  }
+  syncSucceeded(connector.id);
```

Comment:
No need to await this ? If not awaited, might be cool to `.catch` it so we don't get `unhandled promise rejection` (I'm surprised our config lets you have a dangling promise)